models/schema_template_contract_service_graph.go

```diff
@@ -65,3 +65,40 @@ func (graphAttr *SiteContractServiceGraph) ToMap() (map[string]interface{}, erro
 
 	return graphAttrMap, nil
 }
+
+type SiteContractServiceGraphListener struct {
+	Ops   string                 `json:",omitempty"`
+	Path  string                 `json:",omitempty"`
+	Value map[string]interface{} `json:",omitempty"`
+}
+
+func NewSiteContractServiceGraphListener(ops, path, name, protocol, securityPolicy string, port int, certificates []interface{}, rules []interface{}, frontendIpDnMap map[string]string) *SiteContractServiceGraphListener {
+	var listenerMap map[string]interface{}
+	if ops != "remove" {
```

Comment:
I don't think this check is needed when you will use, GetRemovePatchPayload in the resource.